pkg/ottl/contexts/ottlresource/resource_test.go

```diff
@@ -0,0 +1,274 @@
+			}(),
+			newVal: []string{"new"},
+			modified: func(resource pcommon.Resource) {
+				resource.Attributes().PutEmptySlice("arr_str").AppendEmpty().SetStr("new")
+			},
+		},
+		{
+			name: "attributes array bool",
+			path: []ottl.Field{
+				{
+					Name:   "attributes",
+					MapKey: ottltest.Strp("arr_bool"),
+				},
+			},
+			orig: func() pcommon.Slice {
+				val, _ := refResource.Attributes().Get("arr_bool")
+				return val.Slice()
+			}(),
+			newVal: []bool{false},
+			modified: func(resource pcommon.Resource) {
+				resource.Attributes().PutEmptySlice("arr_bool").AppendEmpty().SetBool(false)
+			},
+		},
+		{
+			name: "attributes array int",
+			path: []ottl.Field{
+				{
+					Name:   "attributes",
+					MapKey: ottltest.Strp("arr_int"),
+				},
+			},
+			orig: func() pcommon.Slice {
+				val, _ := refResource.Attributes().Get("arr_int")
+				return val.Slice()
+			}(),
+			newVal: []int64{20},
+			modified: func(resource pcommon.Resource) {
+				resource.Attributes().PutEmptySlice("arr_int").AppendEmpty().SetInt(20)
+			},
+		},
+		{
+			name: "attributes array float",
+			path: []ottl.Field{
+				{
+					Name:   "attributes",
+					MapKey: ottltest.Strp("arr_float"),
+				},
+			},
+			orig: func() pcommon.Slice {
+				val, _ := refResource.Attributes().Get("arr_float")
+				return val.Slice()
+			}(),
+			newVal: []float64{2.0},
+			modified: func(resource pcommon.Resource) {
+				resource.Attributes().PutEmptySlice("arr_float").AppendEmpty().SetDouble(2.0)
+			},
+		},
+		{
+			name: "attributes array bytes",
+			path: []ottl.Field{
+				{
+					Name:   "attributes",
+					MapKey: ottltest.Strp("arr_bytes"),
+				},
+			},
+			orig: func() pcommon.Slice {
+				val, _ := refResource.Attributes().Get("arr_bytes")
+				return val.Slice()
+			}(),
+			newVal: [][]byte{{9, 6, 4}},
+			modified: func(resource pcommon.Resource) {
+				resource.Attributes().PutEmptySlice("arr_bytes").AppendEmpty().SetEmptyBytes().FromRaw([]byte{9, 6, 4})
+			},
+		},
+		{
+			name: "dropped_attributes_count",
+			path: []ottl.Field{
+				{
+					Name: "dropped_attributes_count",
+				},
+			},
+			orig:   int64(10),
+			newVal: int64(20),
+			modified: func(resource pcommon.Resource) {
+				resource.SetDroppedAttributesCount(20)
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			accessor, err := newPathGetSetter(tt.path)
+			assert.NoError(t, err)
+
+			resource := createTelemetry()
+
+			got := accessor.Get(NewTransformContext(resource))
+			assert.Equal(t, tt.orig, got)
+
+			accessor.Set(NewTransformContext(resource), tt.newVal)
+
+			exRes := createTelemetry()
+			tt.modified(exRes)
+
+			assert.Equal(t, exRes, resource)
+		})
+	}
+}
+
+func createTelemetry() pcommon.Resource {
+	resource := pcommon.NewResource()
+
+	resource.Attributes().PutStr("str", "val")
+	resource.Attributes().PutBool("bool", true)
+	resource.Attributes().PutInt("int", 10)
+	resource.Attributes().PutDouble("double", 1.2)
```

Comment:
```suggestion
	resource.Attributes().PutDouble("float", 1.2)
```
Nitpick: floating-point values are referred to with "float" pretty much everywhere else in this file outside of the pdata methods.

Reply:
This pattern is throughout the ottl's tests.  If we want to change it to flout to be consistent with the ottl's message of using float64 I'd suggest we do it in its own PR, fixing all the instances at once.